lib/MirrorCache/Schema/ResultSet/AuditEvent.pm

```diff
@@ -133,4 +137,73 @@ sub mirror_misses {
     return ($id, \%path_country, \@country_list);
 }
 
+sub cleanup_audit_events {
+    my ($self, $app) = @_;
+
+    my $rsource = $self->result_source;
+    my $dbh     = $rsource->schema->storage->dbh;
+
+    my @queries;
+    my $other_time_constraint;
+
+    my %event_patterns = (
+        # system events
+        startup => 'startup',
+        path    => 'path_%',
+        mirror  => 'mirror_%',
+        error   => 'error_%',
+        # user events from the UI
+        user   => 'user_%',
+        server => 'server_%',
+    );
+    # duration in days
+    my %storage_durations = (
+        # system events
+        startup => 14,
+        path    => 14,
+        mirror  => 14,
+        error   => 14,
+        # user events from the UI
+        user   => 90,
+        server => 90,
+        # events not defined above
+        other => 14,
+    );
+
+    for my $event_category (keys %storage_durations) {
+        my $duration_in_days = $storage_durations{$event_category};
+        next unless $duration_in_days;
+
+        my $time_constraint = parsedate("$duration_in_days days ago", PREFER_PAST => 1, DATE_REQUIRED => 1);
+        if (!$time_constraint) {
+            $app->log->warn(
+                "Ignoring invalid storage duration '$duration_in_days' for audit event type '$event_category'.");
+            next;
+        }
+        $time_constraint = localtime($time_constraint)->ymd;
+
+        if ($event_category eq 'other') {
+            $other_time_constraint = $time_constraint;
+            next;
+        }
+
+        my $event_pattern = $event_patterns{$event_category};
+        if (!$event_pattern) {
+            $app->log->warn("Ignoring unknown event type '$event_category'.");
+            next;
+        }
+        push(@queries, {name => {-like => $event_pattern}, dt => {'<' => $time_constraint}});
+    }
+
+    my @pattern_values = values %event_patterns;
+    my $delete_other_query
+      = $other_time_constraint
+      ? $dbh->prepare(
+        'delete from audit_event where dt < ? and ' . join(' and ', map { 'name not like ?' } @pattern_values))
+      : undef;
+
+    $self->search($_)->delete for @queries;
```

Comment:
What do you think about:
1. Add `limit 100000` to the conditions used in delete (hope it is easy to achive)
2. Wrap each delete into `eval`, so if one delete fails - still continue with the rest (I think it will be enough to `$app->log->warn` each error message , plus return count of errors and then log then for last run into `$job->notes`)
3. Add new index `(dt, name)` on audit_event 
4. (optional) maybe add json index on `event_data` as well? We can do it later eventually when needed as well.